2012-04-20  Thomas Schwinge  <thomas@codesourcery.com>

	struct siginfo vs. siginfo_t

	Backport from trunk (but apply to gcc/):

	2012-04-20  Thomas Schwinge  <thomas@codesourcery.com>

	* config/alpha/linux-unwind.h (alpha_fallback_frame_state): Use
	siginfo_t instead of struct siginfo.
	* config/bfin/linux-unwind.h (bfin_fallback_frame_state): Likewise.
	* config/i386/linux-unwind.h (x86_fallback_frame_state): Likewise.
	* config/ia64/linux-unwind.h (ia64_fallback_frame_state)
	(ia64_handle_unwabi): Likewise.
	* config/mips/linux-unwind.h (mips_fallback_frame_state): Likewise.
	* config/pa/linux-unwind.h (pa32_fallback_frame_state): Likewise.
	* config/sh/linux-unwind.h (shmedia_fallback_frame_state)
	(sh_fallback_frame_state): Likewise.
	* config/xtensa/linux-unwind.h (xtensa_fallback_frame_state): Likewise.

--- gcc/config/alpha/linux-unwind.h	(revision 186612)
+++ gcc/config/alpha/linux-unwind.h	(revision 186613)
@@ -49,7 +49,7 @@ alpha_fallback_frame_state (struct _Unwi
   else if (pc[1] == 0x201f015f)		/* lda $0,NR_rt_sigreturn */
     {
       struct rt_sigframe {
-	struct siginfo info;
+	siginfo_t info;
 	struct ucontext uc;
       } *rt_ = context->cfa;
       sc = &rt_->uc.uc_mcontext;
--- gcc/config/i386/linux-unwind.h	(revision 186612)
+++ gcc/config/i386/linux-unwind.h	(revision 186613)
@@ -133,9 +133,9 @@ x86_fallback_frame_state (struct _Unwind
     {
       struct rt_sigframe {
 	int sig;
-	struct siginfo *pinfo;
+	siginfo_t *pinfo;
 	void *puc;
-	struct siginfo info;
+	siginfo_t info;
 	struct ucontext uc;
       } *rt_ = context->cfa;
       /* The void * cast is necessary to avoid an aliasing warning.
--- gcc/config/sh/linux-unwind.h	(revision 186612)
+++ gcc/config/sh/linux-unwind.h	(revision 186613)
@@ -80,9 +81,9 @@ shmedia_fallback_frame_state (struct _Un
 	   && (*(unsigned long *) (pc+11)  == 0x6ff0fff0))
     {
       struct rt_sigframe {
-	struct siginfo *pinfo;
+	siginfo_t *pinfo;
 	void *puc;
-	struct siginfo info;
+	siginfo_t info;
 	struct ucontext uc;
       } *rt_ = context->cfa;
       /* The void * cast is necessary to avoid an aliasing warning.
@@ -179,7 +180,7 @@ sh_fallback_frame_state (struct _Unwind_
 		&& (*(unsigned short *) (pc+14)  == 0x00ad))))
     {
       struct rt_sigframe {
-	struct siginfo info;
+	siginfo_t info;
 	struct ucontext uc;
       } *rt_ = context->cfa;
       /* The void * cast is necessary to avoid an aliasing warning.
--- gcc/config/xtensa/linux-unwind.h	(revision 186612)
+++ gcc/config/xtensa/linux-unwind.h	(revision 186613)
@@ -62,7 +62,7 @@ xtensa_fallback_frame_state (struct _Unw
   struct sigcontext *sc;
 
   struct rt_sigframe {
-    struct siginfo info;
+    siginfo_t info;
     struct ucontext uc;
   } *rt_;
 
--- gcc/config/ia64/linux-unwind.h	(revision 186612)
+++ gcc/config/ia64/linux-unwind.h	(revision 186613)
@@ -47,7 +47,7 @@ ia64_fallback_frame_state (struct _Unwin
       struct sigframe {
 	char scratch[16];
 	unsigned long sig_number;
-	struct siginfo *info;
+	siginfo_t *info;
 	struct sigcontext *sc;
       } *frame_ = (struct sigframe *)context->psp;
       struct sigcontext *sc = frame_->sc;
@@ -137,7 +137,7 @@ ia64_handle_unwabi (struct _Unwind_Conte
       struct sigframe {
 	char scratch[16];
 	unsigned long sig_number;
-	struct siginfo *info;
+	siginfo_t *info;
 	struct sigcontext *sc;
       } *frame = (struct sigframe *)context->psp;
       struct sigcontext *sc = frame->sc;
--- gcc/config/pa/linux-unwind.h	(revision 186612)
+++ gcc/config/pa/linux-unwind.h	(revision 186613)
@@ -63,7 +63,7 @@ pa32_fallback_frame_state (struct _Unwin
   int i;
   struct sigcontext *sc;
   struct rt_sigframe {
-    struct siginfo info;
+    siginfo_t info;
     struct ucontext uc;
   } *frame;
 
--- gcc/config/mips/linux-unwind.h	(revision 186612)
+++ gcc/config/mips/linux-unwind.h	(revision 186613)
@@ -75,7 +76,7 @@ mips_fallback_frame_state (struct _Unwin
       struct rt_sigframe {
 	u_int32_t ass[4];  /* Argument save space for o32.  */
 	u_int32_t trampoline[2];
-	struct siginfo info;
+	siginfo_t info;
 	_sig_ucontext_t uc;
       } *rt_ = context->cfa;
       sc = &rt_->uc.uc_mcontext;
--- gcc/config/bfin/linux-unwind.h	(revision 186612)
+++ gcc/config/bfin/linux-unwind.h	(revision 186613)
@@ -48,10 +48,10 @@ bfin_fallback_frame_state (struct _Unwin
     {
       struct rt_sigframe {
 	int sig;
-	struct siginfo *pinfo;
+	siginfo_t *pinfo;
 	void *puc;
 	char retcode[8];
-	struct siginfo info;
+	siginfo_t info;
 	struct ucontext uc;
       } *rt_ = context->cfa;
 
